src/backend/database_models/conversation.py

```diff
@@ -18,9 +19,15 @@ class Conversation(Base):
 
     text_messages: Mapped[List["Message"]] = relationship()
     files: Mapped[List["File"]] = relationship()
+    agent_id: Mapped[str] = mapped_column(
+        ForeignKey("agents.id", ondelete="CASCADE"), nullable=True
+    )
 
     @property
     def messages(self):
         return sorted(self.text_messages, key=lambda x: x.position)
 
-    __table_args__ = (Index("conversation_user_id", user_id),)
+    __table_args__ = (
+        Index("conversation_user_id", user_id),
+        Index("conversation_agent_id", agent_id),
```

Comment:
If conversations are filtered by `agent_id` only while filtering by `user_id`, you could get more benefits by adding a composite index of both `user_id + agent_id`